src/internal/kvparse/kvgrammar/kvgrammar.go

```diff
@@ -0,0 +1,179 @@
+package kvgrammar
+
+import (
+	"strconv"
+	"strings"
+
+	"github.com/alecthomas/participle/v2/lexer"
```

Comment:
Looking at your proposed grammar, I believe that it is simple enough that a hand-written parser could be this library-using version, which would not require adding an external dependency. I know, I know, “I’ll just write a parse” is one of the famous last phrases before a security exploit …

OTOH, it wouldn’t have nice sugar like pretty-printing the BNF.